apps/E2E/src/common/BasePage.ts

```diff
@@ -72,6 +82,48 @@ export abstract class BasePage {
   /**************** UI Element Interaction Methods ******************/
   /******************************************************************/
 
+  /**
+   * For any given page object, this method automates:
+   *  - Navigating to the page by clicking its component button
+   *  - Waiting for the component page to load
+   *  - Revealing its E2E components IF a boolean flag is passed into the method
+   *
+   * This also contains error checking through its waiters, removing the extra `expect()` calls during test setup.
+   */
+  async navigateToPageAndLoadTests(showE2ESection = false) {
+    // Desktop platforms automatically scroll to a page's navigation button - this extra step is purely for mobile platforms.
+    if (this.platform === MobilePlatform.Android || this.platform === MobilePlatform.iOS) {
+      await this.mobileScrollToComponentButton();
+    }
+
+    await (await this._pageButton).click();
+    await this.waitForPageDisplayed();
+
+    if (showE2ESection) {
+      await this.enableE2ETesterMode();
+    }
+  }
+
+  async enableE2ETesterMode(): Promise<void> {
+    const e2eSwitch = await this._e2eSwitch;
+    switch (this.platform) {
+      case MobilePlatform.Android:
+        if ((await e2eSwitch.getAttribute(AndroidAttribute.Checked)) === 'false') {
+          await e2eSwitch.click();
+          await this.waitForCondition(
+            async () => (await e2eSwitch.getAttribute(AndroidAttribute.Checked)) === 'true',
+            'Clicked the E2E Mode Switch, but it failed to toggle.',
+          );
+        }
+        break;
+      default:
+        if (!(await e2eSwitch.isSelected())) {
+          await e2eSwitch.click();
+          await this.waitForCondition(async () => e2eSwitch.isSelected(), 'Clicked the E2E Mode Switch, but it failed to toggle.');
+        }
```

Comment:
I think we should also add a wait outside of the case statement that validates the E2E section is visibly shown correctly.

This might require adding another E2E testing constant on the title of the E2E section